frontend/src/metabase/querying/components/TimeseriesChrome/TimeseriesBucketPicker/TimeseriesBucketPicker.tsx

```diff
@@ -14,97 +14,97 @@ interface TimeseriesBucketPickerProps {
-  const availableItems = useMemo(
-    () => getAvailableItems(query, stageIndex, column),
-    [query, stageIndex, column],
-  );
-
-  const [isOpen, setIsOpen] = useState(false);
-  const [isExpanded, setIsExpanded] = useState(
-    isInitiallyExpanded(availableItems, selectedItem),
-  );
-
-  const handleOpen = useCallback(() => {
-    setIsOpen(true);
-  }, [setIsOpen]);
+  const [isOpened, setIsOpened] = useState(false);
 
-  const handleClose = useCallback(() => {
-    setIsOpen(false);
-    setIsExpanded(false);
-  }, [setIsOpen]);
+  const columnBucketInfo = useMemo(() => {
+    const bucket = Lib.temporalBucket(column);
+    return bucket ? Lib.displayInfo(query, stageIndex, bucket) : undefined;
+  }, [query, stageIndex, column]);
 
-  const handleChange = (bucket: Lib.Bucket | null) => {
-    onChange(Lib.withTemporalBucket(column, bucket));
-    setIsOpen(false);
-    setIsExpanded(false);
+  const handleChange = (newColumn: Lib.ColumnMetadata) => {
+    onChange(newColumn);
+    setIsOpened(false);
   };
 
-  const handleExpand = useCallback((evt: React.MouseEvent) => {
-    evt.stopPropagation();
-    evt.preventDefault();
-    setIsExpanded(true);
-  }, []);
-
-  const canExpand = availableItems.length > INITIALLY_VISIBLE_ITEMS_COUNT;
-  const hasMoreButton = canExpand && !isExpanded;
-  const visibleItems = hasMoreButton
-    ? availableItems.slice(0, INITIALLY_VISIBLE_ITEMS_COUNT)
-    : availableItems;
-
   return (
-    <Menu
-      closeOnItemClick={false}
-      opened={isOpen}
-      onOpen={handleOpen}
-      onClose={handleClose}
-    >
-      <Menu.Target>
+    <Popover opened={isOpened} onChange={setIsOpened}>
+      <Popover.Target>
         <Button
           rightIcon={<Icon name="chevrondown" />}
           data-testid="timeseries-bucket-button"
+          onClick={() => setIsOpened(!isOpened)}
         >
-          {selectedItem.name}
+          {columnBucketInfo ? columnBucketInfo.displayName : t`Unbinned`}
         </Button>
-      </Menu.Target>
-      <Menu.Dropdown>
-        {visibleItems.map((option, index) => (
-          <Menu.Item key={index} onClick={() => handleChange(option.bucket)}>
-            {option.name}
-          </Menu.Item>
-        ))}
-
-        {hasMoreButton && (
-          <Menu.Item onClick={handleExpand}>
-            <Text color="brand">{t`More…`}</Text>
-          </Menu.Item>
-        )}
-      </Menu.Dropdown>
-    </Menu>
+      </Popover.Target>
+      <Popover.Dropdown>
+        <TimeseriesBucketDropdown
+          query={query}
+          stageIndex={stageIndex}
+          column={column}
+          columnBucketInfo={columnBucketInfo}
+          onChange={handleChange}
+        />
+      </Popover.Dropdown>
+    </Popover>
   );
 }
 
-function isInitiallyExpanded(items: BucketItem[], selectedItem: BucketItem) {
-  const canExpand = items.length > INITIALLY_VISIBLE_ITEMS_COUNT;
-  if (!canExpand || !selectedItem) {
-    return false;
-  }
+interface TimeseriesBucketDropdownProps {
```

Comment:
It's always a good practice to any calls (especially MBQL lib) when the dropdown is actually opened.